bases/renku_data_services/background_jobs/utils.py

```diff
@@ -0,0 +1,60 @@
+"""Different utility functions for background jobs."""
+
+import logging
+
+from authzed.api.v1.permission_service_pb2 import ReadRelationshipsRequest, RelationshipFilter
+
+from renku_data_services.authz.authz import Authz, ResourceType, _Relation
+from renku_data_services.background_jobs.config import SyncConfig
+from renku_data_services.message_queue.avro_models.io.renku.events import v2
+from renku_data_services.message_queue.converters import EventConverter
+
+
+async def sync_user_namespaces(config: SyncConfig) -> None:
+    """Lists all user namespaces in the database and adds them to Authzed and the event queue."""
+    authz = Authz(config.authz_config)
+    user_namespaces = await config.group_repo._get_user_namespaces()
```

Comment:
I think it'd be good to get namespaces in batches (You can have an AsyncGenerator that yields batches of namespaces and use it with `async for ...`) otherwise this might have a pretty big result set as time goes on.

Reply:
So I used a stream from sqlalchemy which is supposed to gradually stream the results from the server. And I used async for. So there is no need to batch this - the results will come gradually.